Modules/_ctypes/_ctypes.c

```diff
@@ -3094,6 +3115,7 @@ static PyType_Spec pycdata_spec = {
 static int
 PyCData_MallocBuffer(CDataObject *obj, StgInfo *info)
 {
+    // We don't have to lock in this function
```

Comment:
I think it's clear that you don't need to lock it if you don't have locks. However, it'd be better if you explain why.